src/Cargo.toml

```diff
@@ -44,9 +44,9 @@ members = [
 # MSVC when running the compile-fail test suite when a should-fail test panics.
 # But hey if this is removed and it gets past the bots, sounds good to me.
 [profile.release]
-opt-level = 2
+opt-level = 3
```

Comment:
This is likely still unsafe. See #48204.

Reply:
LLVM was updated to 6.0 since then, so that bug may be fixed. I know that at least one codegen bug disappeared with that update.

Reply:
#48204 was tested on [2018 Feb 25th](https://github.com/rust-lang/rust/pull/48204#issuecomment-368290008)  where [LLVM was already upgraded to 6.0](https://github.com/rust-lang/rust/pull/47828#issuecomment-364628215) (2018 Feb 10th).

Reply:
LLVM 6.0 was released Mar 8, 2018.

Reply:
Although #47828 isn't the most up-to-date LLVM 6 branch and #48782 corresponds to the actual release version, there's nothing guaranteeing -O3 would become safe.

Please split this change into a separate PR anyway. Given the bad [track record](https://github.com/rust-lang/rust/pull/48204#issuecomment-367787994), I wouldn't want to lump this together with other things here (especially you're including so many Windows changes).

Reply:
I agree with @kennytm that this I think is best left to a separate PR so we can track it independently in terms of perf, regressions, etc.

Reply:
I made a separate PR for this.